qutebrowser/browser/browsertab.py

```diff
@@ -133,12 +134,97 @@ class TabData:
     fullscreen = attr.ib(False)  # type: bool
     netrc_used = attr.ib(False)  # type: bool
     input_mode = attr.ib(usertypes.KeyMode.normal)  # type: usertypes.KeyMode
+    splitter = attr.ib(None)
 
     def should_show_icon(self) -> bool:
         return (config.val.tabs.favicons.show == 'always' or
                 config.val.tabs.favicons.show == 'pinned' and self.pinned)
 
 
+class InspectorSplitter(QSplitter):
+
+    """Allows putting an inspector inside the tab.
+
+    Attributes:
+        _main_idx: index of the main webview widget
+        _position: position of the inspector (right/left/top/bottom)
+        _preferred_size: the preferred size of the inpector widget in pixels
+    """
+
+    _main_idx = 0
+    _position = None
+    _preferred_size = None
+
+    def __init__(self, main_webview):
+        super().__init__()
+        self.addWidget(main_webview)
+        self.splitterMoved.connect(self._onSplitterMoved)
+
+    def set_inspector(self, inspector, position):
+        """Set the position of the inspector."""
+        assert position in ['right', 'left', 'top', 'bottom']
+        self._main_idx = 0 if position in ['right', 'bottom'] else 1
+        self.setStretchFactor(self._main_idx, 1)
+        self.setStretchFactor(self._inspector_idx(), 0)
+        self.setOrientation(Qt.Horizontal if position in ['right', 'left']
+                            else Qt.Vertical)
+        self.insertWidget(self._inspector_idx(), inspector)
+        self._position = position
+        self._load_preferred_size()
+        self._adjust_size()
+
+    def _save_preferred_size(self):
+        """Save the preferred size of the inspector widget."""
+        configfiles.state['geometry']['inspector_' + self._position] = str(
+            self._preferred_size)
+
+    def _load_preferred_size(self):
+        """Load the preferred size of the inspector widget."""
+        if self.orientation() == Qt.Horizontal:
+            full = self.width()
+        else:
+            full = self.height()
+        self._preferred_size = max(300, full / 2)
```

Comment:
Why 300? I don't think this should be a magic number

Reply:
I agree that hardcoding numbers is not that beautiful.
The idea is that the inspector is of not much use when it is really small so it should use more than  half the width/height of a very small window.